### PR TITLE
Conditional Logic UI

### LINKED ISSUE #321: Implement Conditional Logic on frontend
- [x] Allow expanding criteria to logical list
- [x] Allow contracting logical list to criteria
- [x] Allow deleting criteria
- [x] Allow deleting logical list
- [x] Deleting last criteria disables condition editor
- [ ] Highlight pages in sidebar containing operand elements
- [ ] Show preview of condition in page list
- [x] Allow adding criteria to list
- [ ] Auto-complete for elements
- [ ] Auto-complete for value (if element is a choice-based element)
- [ ] First page should have no options to create conditional logic
- [ ] If page is moved to first page, condition should be removed

### PR DESCRIPTION
**Pull request by Trinovantes**
Closes #321 

screenshot from 2016-02-24 01 00 41

### CODE REVIEW

File: src-backbone/app/js/views/builder/pageConditions/conditionalNodeView.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        if (!this.model.isRootNode()) {
+            this.parentNodeView.render();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Criteria Node
+    //--------------------------------------------------------------------------
+
+    _onCriteriaNegationChanged: function(event) {
```
Comment (Kapin) [inline src-backbone/app/js/views/builder/pageConditions/conditionalNodeView.js]: debounce this please

File: src-backbone/app/js/views/builder/pageConditions/conditionalNodeView.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        this.model.saveRootShowIf();
+    },
+
+    _onCriteriaComparatorChanged: function(event) {
+        let operand = this.ui.criteriaComparator.val();
+        this.model.set('node_type', operand);
+        this.model.saveRootShowIf();
+    },
+
+    _onCriteriaElementChanged: function(event) {
```
Comment (Kapin) [inline src-backbone/app/js/views/builder/pageConditions/conditionalNodeView.js]: debounce

File: src-backbone/app/js/views/builder/pageConditions/conditionalNodeView.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+        this.model.delete();
+
+        if (!this.model.isRootNode()) {
+            this.parentNodeView.render();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Criteria Node
```
Comment (Kapin) [inline src-backbone/app/js/views/builder/pageConditions/conditionalNodeView.js]: All of these need debouncing

**Comment by Kapin** [comment]
Criteria nodes need to be negated as well. I don't see that being handled in the screenshot above.

e.g. 

<not>
  <Criteria.../>
</not>

**Comment by Trinovantes** [comment]
It's handled by the dropdown in the criteria node

**Comment by Kapin** [comment]
I think it looks fine to me. I'd like everyone to take a look at it though.

File: src-backbone/app/js/models/conditionals/conditionalNode.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    },
+
+    isCriteriaNode: function() {
+        let nodeType = this.get('node_type');
+        if (CRITERIA_NODE_TYPES.includes(nodeType)) {
+            return true;
+        } else if (LOGICAL_NODE_TYPES.includes(nodeType)) {
+            return false;
+        } else {
+            throw ReferenceError('Unknonwn conditional node type', nodeType);
```
Comment (Ommy) [inline src-backbone/app/js/models/conditionals/conditionalNode.js]: typo `Unknonwn`

**Comment by Ommy** [comment]
lgtm after comment

**Comment by connorcimowsky** [comment]
lgtm!

**Pull request merged**